crates/miden-lib/src/note/well_known_note.rs

```diff
@@ -157,4 +163,242 @@ impl WellKnownNote {
+                let timelock_height = u32::try_from(note_inputs[3]).map_err(|_err| {
+                    StaticAnalysisError::new("timelock block height should be a u32")
+                })?;
+
+                let current_block_height = block_ref.as_u32();
+
+                // Handle the case when current block height is greater or equal to the timelock
+                // height and reclaim height (first row in the table).
+                //
+                // Return None: both sender and receiver accounts can consume the note
+                if current_block_height >= reclaim_height.max(timelock_height) {
+                    return Ok(None);
+                }
+
+                // Handle the case when timelock height is greater than the current block height and
+                // reclaim height (second row in the table).
+                //
+                // Return the timelock height: neither sender, nor the target cannot consume the
+                // note.
+                if timelock_height > current_block_height.max(reclaim_height) {
+                    return Ok(Some(NoteConsumptionStatus::ConsumableAfter(BlockNumber::from(
+                        timelock_height,
+                    ))));
+                }
+
+                // Handle the case when the target account is the sender and the reclaim height is
+                // strictly greater than the current block height and greater or equal to the
+                // timelock height (`Sender` column of the last two rows in the table).
+                //
+                // Return the reclaim height since for the sender it is only important to know that
+                // the reclaim height is not reached yet and the reclaim height is greater than the
+                // timelock height (we should wait for the reclaim anyway).
+                if target_account_id == note.metadata().sender()
+                    && reclaim_height > current_block_height
+                    && reclaim_height >= timelock_height
+                {
+                    Ok(Some(NoteConsumptionStatus::ConsumableAfter(BlockNumber::from(
+                        reclaim_height,
+                    ))))
+                } else {
+                    // Receiver doesn't care what is the height of the reclaim block, so it should
+                    // be only checked whether the timelock height is greater than the current block
+                    // height or not (`Receiver` column of the last two rows in the table)
+                    if timelock_height > current_block_height {
+                        Ok(Some(NoteConsumptionStatus::ConsumableAfter(BlockNumber::from(
+                            timelock_height,
+                        ))))
+                    } else {
+                        Ok(None)
+                    }
+                }
+            },
+
+            // if the note is `SWAP`, try to execute it to determine whether it could be consumed
+            _ => Ok(None),
```

Comment:
Nit: We're not executing anything here and this assumes the knowledge that this API is being consumed by the note checker, so would be good to rephrase this to clarify.